src/main/java/fr/kungfunantes/backend/resource/ExerciseResource.java

```diff
@@ -2,22 +2,61 @@
 
 import fr.kungfunantes.backend.model.exercise.Exercise;
 import fr.kungfunantes.backend.repository.ExerciseRepository;
+import fr.kungfunantes.backend.repository.ExerciseCriteriaRepository;
+import fr.kungfunantes.backend.repository.ExerciseRoundRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
+import java.util.Optional;
 
 @RestController
 public class ExerciseResource {
 
 	@Autowired
 	private ExerciseRepository exerciseRepository;
 
-    @GetMapping("/exercises")
-    @ResponseBody
-    public List<Exercise> all() {
-        return exerciseRepository.findAll();
+	@Autowired
+	private ExerciseCriteriaRepository exerciseCriteriaRepository;
+
+	@Autowired
+	private ExerciseRoundRepository exerciseRoundRepository;
+
+	@GetMapping("/exercises")
+	@ResponseBody
+	public List<Exercise> all() {
+			return exerciseRepository.findAll();
 	}
+
+	@GetMapping("/exercises/{id}")
+	@ResponseBody
+	public Optional<Exercise> byId(@PathVariable("id") Long id) {
+			return exerciseRepository.findById(id);
+	}
+
+	@DeleteMapping("/exercises/{id}")
+	@ResponseStatus(HttpStatus.OK)
+	@Transactional
+	public void delete(@PathVariable("id") Long id) {
+		System.out.println("COUCOU");
```

Comment:
You should avoid that ;)
I will merge this, but please remove this in the next PR